mlir/lib/IR/Types.cpp

```diff
@@ -63,6 +63,8 @@ bool Type::isF128() const { return llvm::isa<Float128Type>(*this); }
 
 bool Type::isIndex() const { return llvm::isa<IndexType>(*this); }
 
+bool Type::isFloat() const { return llvm::isa<FloatType>(*this); }
+
```

Comment:
This can be seen as fitting here for consistency, however I wonder if the desirable thing wouldn't instead be to remove all the other `isXXXX` from the generic `Type` class?